src/NATS.Client.Core/NatsConnection.cs

```diff
@@ -289,6 +288,65 @@ internal ValueTask UnsubscribeAsync(int sid)
         return default;
     }
 
+    private static NatsOpts ReadUserInfoFromConnectionString(NatsOpts opts)
+    {
+        var first = true;
+
+        var natsUris = opts.GetSeedUris(suppressRandomization: true);
+        var maskedUris = new List<string>(natsUris.Length);
+
+        foreach (var natsUri in natsUris)
+        {
+            var uriBuilder = new UriBuilder(natsUri.Uri);
+
+            if (uriBuilder.UserName is { Length: > 0 } username)
+            {
+                if (uriBuilder.Password is { Length: > 0 } password)
+                {
+                    if (first)
+                    {
+                        first = false;
+
+                        opts = opts with
+                        {
+                            AuthOpts = opts.AuthOpts with
+                            {
+                                Username = uriBuilder.UserName,
+                                Password = uriBuilder.Password,
+                            },
+                        };
+                    }
+
+                    uriBuilder.Password = "***"; // to redact the password from logs
```

Comment:
username needs to be set as well in case they are different.

Reply:
Sorry, I don't understand what you mean? Username should also be redacted when using password?

Reply:
no in case a different username is provided e.g. on the second url logs still should show the first username which is actually being used to connect.

Reply:
I hope I have fixed it how you meant it